src/client/nativeCommunication.ts

```diff
@@ -127,7 +110,73 @@ function slideshow(): void {
         }));
 }
 
+function getVideoSlots(): VideoSlot[] {
+    const videoSlots = document.querySelectorAll('.native-video');
+
+    if (!videoSlots) {
+        return [];
+    }
+
+    return Array.from(videoSlots).map(videoSlot => {
+        const slotPosition = videoSlot.getBoundingClientRect();
+        return new VideoSlot({
+            rect: getRect(slotPosition),
+            videoId: videoSlot.getAttribute('data-videoId') ?? '',
+            posterUrl: videoSlot.getAttribute('data-posterUrl') ?? '',
+            duration: parseInt(videoSlot.getAttribute('data-duration') ?? '0')
+        })
+    });
+}
+
+function videos(): void {
+    const videoSlots = getVideoSlots();
+    if (videoSlots.length) {
+        videoClient.insertVideos(videoSlots);
+    }
+}
+
+function reportNativeElementPositionChanges(): void {
+    let adSlots = getAdSlots();
+    let videoSlots = getVideoSlots();
+
+    const targetNode = document.querySelector('html') as Node;
+    const config: MutationObserverInit = {
+        childList: true,
+        subtree: true,
+        attributeFilter: ["style"]
+    };
+    const callback = function(): void {
+        const currentAdSlots = getAdSlots();
+        const currentVideoSlots = getVideoSlots();
+
+        if (JSON.stringify(adSlots) !== JSON.stringify(currentAdSlots)) {
```

Comment:
Is that a performance risk right here?
Is this executed each time there's a scroll, or each time there's a re-layout?

Reply:
At the moment it's only run when a `style` tag is changed (font resizing) but yeah we need to keep an eye on this and possibly debounce it when we extend the mutation observer to watch for other changes.

Reply:
I may have missed something, but why `JSON.stringify` and compare the strings? Why not compare the arrays directly?

Reply:
Maybe we could compare all the values now so stringify isn't necessary... but a direct comparison doesn't work

[1, 2, 3] === [1, 2, 3]
false